src/Reports/JsonDataSource.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace iEducar\Reports;
+
+use Exception;
+use Portabilis_Utils_Database;
+
+trait JsonDataSource
+{
+    /**
+     * @inheritdoc
+     */
+    public function useJson()
+    {
+        return true;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getJsonData()
+    {
+        $queryMainReport = $this->getSqlMainReport();
```

Comment:
Acha que seria válido ter dois métodos aqui, um pra retornar a query e outro para os parâmetros? Mesmo que em um primeiro momento retorne a query pronta e um array vazio

Falo isso porque acho que facilitaria trocar por PDO no futuro e fazer bind dos parâmetros.

ping @edersoares

Reply:
Sim @munizeverton o ideal é refatorar as queries e fazermos o bind dos parâmetros, porém como o tempo estava curto, foi optado por apenas remover a query do arquivo `.jrxml` e modificá-lo o menos possível com o intuito de não quebrar nada.

A partir de agora, o céu é o limite. Podemos começar a trabalhar com PDO, ORM ou uma camada de repositório para a geração destes dados. Há inúmeras possibilidades.

Acho que podemos fazer esta mudança gradativamente com o apoio da comunidade e logo que começarmos a refatorar algumas queries para melhor desempenho.